src/google/protobuf/stubs/time.cc

```diff
@@ -262,10 +262,10 @@ bool DateTimeToSeconds(const DateTime& time, int64* seconds) {
 }
 
 void GetCurrentTime(int64* seconds, int32* nanos) {
-  // TODO(xiaofeng): Improve the accuracy of this implementation (or just
-  // remove this method from protobuf).
-  *seconds = time(nullptr);
-  *nanos = 0;
+  auto now = std::chrono::high_resolution_clock::now().time_since_epoch();
+  auto secs = std::chrono::duration_cast<std::chrono::seconds>(now);
+  *seconds = secs.count();
+  *nanos = std::chrono::duration_cast<std::chrono::nanoseconds>(now - secs).count();
```

Comment:
Somehow this returns wildly different values for me. It works on GCC with x86, but when compiled on Clang for Apple Silicon it shows something wildly different.
TimeUtil::TimestampToMilliseconds(TimeUtil::GetCurrentTime());
Clang (Apple Si) = 573427268 ms
GCC (X86) = 1642781354185 ms (Received over GRPC)

The delta should be < 100 ms, but it seems pretty big